addon/components/course-materials.js

```diff
@@ -1,126 +1,156 @@
 import Component from '@ember/component';
-import { Promise as RSVPPromise, map } from 'rsvp';
 import EmberObject, { computed } from '@ember/object';
-import { isPresent, isEmpty } from '@ember/utils';
-import SortableTable from 'ilios-common/mixins/sortable-table';
+import { reads } from '@ember/object/computed';
+import { isEmpty, isPresent } from '@ember/utils';
+import { all, task, timeout } from 'ember-concurrency';
 import layout from '../templates/components/course-materials';
+import DS from 'ember-data';
 
-export default Component.extend(SortableTable, {
+const { PromiseArray } = DS;
+
+const DEBOUNCE_DELAY = 250;
+
+export default Component.extend({
   layout,
-  course: null,
-  sortBy: null,
+
   classNames: ['course-materials'],
-  typesWithUrl: null,
-  filter: '',
-  sessionLearningMaterials: computed('course.sessions.[]', function(){
-    const course = this.get('course');
-    return new RSVPPromise(resolve => {
-      course.get('sessions').then(sessions => {
-        map(sessions.toArray(), session => {
-          return session.get('learningMaterials');
-        }).then(learningMaterials => {
-          let flat = learningMaterials.reduce((flattened, obj) => {
-            return flattened.pushObjects(obj.toArray());
-          }, []);
-
-          resolve(flat);
-        });
-      });
-    });
+
+  clmQuery: '',
+  course: null,
+  courseSort: null,
+  sessionSort: null,
+  slmQuery: '',
+  onClmSort() {},
```

Comment:
Am I missing where these two methods are used? What is their purpose?

Reply:
The empty functions are used to signify functions are being passed in from parent component, just like a normal attribute. It's a common pattern.

Example:
```
hbs
{{child-component isLoading=isLoading onChange=(action "whatever")}}

js
isLoading: null,
onChange() {}
```

Reply:
What I'm not seeing is where these are actually called in this component. Did the name change?

Reply:
I changed them to `onCourseSort` and `onSessionSort`.